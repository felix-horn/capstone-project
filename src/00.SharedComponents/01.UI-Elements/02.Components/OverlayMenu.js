import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import Menu from './Menu'

OverlayMenu.propTypes = {
  toggleMenu: PropTypes.func.isRequired,
  deleteShop: PropTypes.func.isRequired,
}

export default function OverlayMenu({ toggleMenu, deleteShop }) {
  return (
    <ClickableMenuBackground onClick={toggleMenu} data-testid="background">
      <MenuStyled deleteShop={deleteShop} />
    </ClickableMenuBackground>
  )
}

const ClickableMenuBackground = styled.div`
  position: absolute;
  z-index: var(--z-index-overlay);
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
`

const MenuStyled = styled(Menu)`
  position: absolute;
  z-index: var(--z-index-item-on-overlay);
  top: 45px;
  right: 0;
`
