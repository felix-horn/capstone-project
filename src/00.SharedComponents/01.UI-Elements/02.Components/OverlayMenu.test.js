/* eslint-disable no-undef */
import user from '@testing-library/user-event'
import OverlayMenu from './OverlayMenu'

describe('OverlayMenu', () => {
  it('renders correctly', () => {
    const toggleMenuMock = jest.fn()
    const deleteShopMock = jest.fn()
    const { container } = renderWithRouter(
      <OverlayMenu toggleMenu={toggleMenuMock} deleteShop={deleteShopMock} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('calls function deleteShop', () => {
    const toggleMenuMock = jest.fn()
    const deleteShopMock = jest.fn()
    const { getByTestId } = renderWithRouter(
      <OverlayMenu toggleMenu={toggleMenuMock} deleteShop={deleteShopMock} />
    )
    const deleteButton = getByTestId('delete-button')
    user.click(deleteButton)
    expect(deleteShopMock).toHaveBeenCalledTimes(1)
  })

  it('calls function toggleMenu', () => {
    const toggleMenuMock = jest.fn()
    const deleteShopMock = jest.fn()
    const { getByTestId } = renderWithRouter(
      <OverlayMenu toggleMenu={toggleMenuMock} deleteShop={deleteShopMock} />
    )
    const background = getByTestId('background')
    user.click(background)
    expect(toggleMenuMock).toHaveBeenCalledTimes(1)
  })
})
