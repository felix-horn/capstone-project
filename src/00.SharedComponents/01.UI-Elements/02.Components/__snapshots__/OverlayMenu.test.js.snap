// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlayMenu renders correctly 1`] = `
.c1 {
  box-shadow: var(--light-box-shadow);
  border-radius: 3px;
  background-color: var(--white);
  padding: 20px;
  display: grid;
  justify-items: start;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--warning-red);
}

.c0 {
  position: absolute;
  z-index: var(--z-index-overlay);
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
}

.c2 {
  position: absolute;
  z-index: var(--z-index-item-on-overlay);
  top: 45px;
  right: 0;
}

<div
  class="c0"
  data-testid="background"
>
  <div
    class="c1 c2"
  >
    <a
      aria-current="page"
      class="c3 active"
      data-testid="delete-button"
      href="/"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
      Liste unwiderruflich l√∂schen
    </a>
  </div>
</div>
`;
